Apply linter suggestions for cleaner code

Lingepumpe opened the pull request:
This PR is on top of the FitSDK 21.141.00 branch.

In this PR additionally to the SDK update, I fixed issues that the linter (clippy) was pointing out. This also involved fixes in autogenerated `profile/decode.rs` and `profile/field_types.rs` files - for this I adapted the code that generates them. Additionally I added `cargo clippy --fix` and `cargo fmt` to the `update_profile.sh` shell script, as for `profile/field_types.rs` this leads to a nice collapsing of huge match "|" cascades into simple ranges, which is much more readable (and maybe even faster, I am not sure what the compiler does).

stadelmanma commented:
Thanks for looking at this, I'm fine with the clippy fixes in the main "human written" code base, those are good. However, I don't think I want to apply them to the auto-generated code at the tail end of the shell script. Clippy re-writes things in a nontrivial way at times, and that will make it harder to track down bugs since the code we see getting written in the generation step doesn't exist 1:1 in the code being run anymore.

That being said, using clippy to suggest ways we can improve code generation does seem like a very good idea, especially if they would be trivial to implement like where it changed `.map().flatten()` to an `.and_then()`

Also I call "rustfmt" via code in the main script, https://github.com/stadelmanma/fitparse-rs/blob/1fdba0560cabacdda4e3a69bc5688f3567d50d65/generate-fit-profile/src/main.rs#L31 so the extra call in the shell script isn't needed. But maybe it was the application of clippy that required that to be done again.

stadelmanma commented:
I think for now I'd like to drop any automatic code changes (other than the programmatically called rustfmt), so more or less removing the changes to `bin/update_profile.sh`. The profile code is impossible to fully validate at this time so I'd like to keep it's creation as straight forward as possible instead of adding another layer of complexity.

I love the other changes, and wouldn't be opposed to updating the codegen script to create the desired output for `matches!` on large ranges directly. Same for anything else suggested by clippy that makes sense, although I'm not sure how far down the rabbit hole we should go, since those files aren't really meant to be "read" by humans.

Lingepumpe commented:
As requested, I have dropped the automatic code changes (clippy fix + cargo fmt) from `update_profile.sh`. Instead, I changed the code that generates `profile/field_types.rs` to detect ranges and generate the first..=last syntax in this case.

Also, I rebased the branch on master, as the FitSDK 21.141.00 branch was merged.

stadelmanma commented:
This looks great now, I'll do some local testing probably tomorrow and get it merged in!

stadelmanma commented:
Thanks!
